core/contracts/primitive_stream.kf

```diff
@@ -333,6 +333,13 @@ procedure get_index($date_from text, $date_to text, $frozen_at int, $base_date t
 
 // get_base_value returns the first nearest value of the primitive stream before the given date
 procedure get_base_value($base_date text, $frozen_at int) private view returns (value decimal(36,18)) {
+    // Coalesce $base_date to '2010-01-01' if it's null or empty.
+    // A valid date is required for the queries since we perform both
+    // > and <= comparisons to fetch records within the date range.
+    if $base_date is null OR $base_date = '' {
+        $base_date := '2010-01-01';
```

Comment:
if it's null, isn't better to return the first value ever (with frozen at), instead of arbitrarily following `2010` truflation specificities?

Reply:
Wouldn't it match the methodology then?
if that's the case, then we need to tell Frontend to give base_date for every query

Reply:
but we already push from 2010 only, then it matches, as we previously didn't have how to specify a `base_date`, right? see that prior to this PR we already didn't use 2010 as a default anywhere in the contract -- it would change the behavior

Reply:
I think that even though we want to pursue a default value for base date, it should be specifically at `tsn-data-provider` repo contract (which is more truflation as a data provider specific)

Reply:
> but we already push data from 2010 only, then it matches. As we previously didn't have how to specify a base_date, right?

Yes, that's correct. But in the methodology document, it is possible that the base date uses a value before `2010`. This is what happening with the Argentina index, hence we must start to push the record even before `2010`. I think reducing it to `2009` would be good since there might be a stream on yearly records.

Reply:
alright, I think it is good for first ever value